### DISCUSSION TITLE
bug: Settings>Advanced Parameters>Context Length

### DISCUSSION THREAD

**Discussion by edwardochoaphd** [open-webui/open-webui#2147] [General]
# Bug Report

## Description

**Bug Summary:**
OpenWebUI settings appear to be adding an extra zero to Context Length
may be related related to [feature request](https://github.com/open-webui/open-webui/issues/1952)?...
![extraZeroIssue](https://github.com/open-webui/open-webui/assets/20013196/73face1b-1ac0-4ca3-9d5d-8661c2040b9a)

**Steps to Reproduce:**
Set context length then read logfile

**Expected Behavior:**
No change to user's context length setting

**Actual Behavior:**
an extra zero appears to be added to context length

## Environment

- **Open WebUI Version:** 0.1.123
- **Ollama (if applicable):** 0.1.34

- **Operating System:** RHEL8
- **Browser (if applicable):**Chrome

## Reproduction Details
Set context length then read logfile. these are 2 logs examples showing my context length had an extra zero added
- time=2024-05-08T20:12:24.868Z level=WARN source=memory.go:17 msg="requested context length is greater than model max context length" requested=81920 model=65536
- time=2024-05-08T05:24:12.168Z level=WARN source=memory.go:17 msg="requested context length is greater than model max context length" requested=20480 model=8192

**Confirmation:**

- [ x] I have read and followed all the instructions provided in the README.md.
- [ x] I am on the latest version of both Open WebUI and Ollama.
- [ x] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots
see attached potentially related issue snapshot with extra zero highlighted

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
see attached

## Installation Method

docker

## Additional Information

see above

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by edwardochoaphd** [comment]
appears there may be a 2nd (related?) issue/bug with Settings>Advanced Parameters>Max Tokens
when set to 2048... in a conversation, getting an extra zero getting added here too?... see below two examples of 20480 which should have been 2048
...
time=2024-05-09T14:33:27.120Z level=WARN source=server.go:77 msg="requested context length is greater than the model's training context window size" requested=20480 "training size"=4096
...
.................................................................................................
llama_new_context_with_model: n_ctx = 20480

**Comment by tjbck** [comment]
<img width="633" alt="image" src="https://github.com/open-webui/open-webui/assets/25473318/913353b1-1fff-4733-891f-c2bfae3caa7f">

We can verify we're sending the right value to Ollama by inspecting the browser dev tool network tab, my guess is this might be an issue on Ollama-end.

**Reply by edwardochoaphd** [reply]
thx @tjbck for your quick reply/checking on this!

**Reply by edwardochoaphd** [reply]
submitted https://github.com/ollama/ollama/issues/4288

**Comment by iukea1** [comment]
> <img width="633" alt="image" src="https://github.com/open-webui/open-webui/assets/25473318/913353b1-1fff-4733-891f-c2bfae3caa7f">
> 
> We can verify we're sending the right value to Ollama by inspecting the browser dev tool network tab, my guess is this might be an issue on Ollama-end.

😎😎😎😎😎😎 Thank you so much for helping troubleshoot this.

**Reply by edwardochoaphd** [reply]
in case this might help others, using OpenWebUI settings, this seems to help while the ollama devs (TBD timeframe) work on the bug...
![image](https://github.com/open-webui/open-webui/assets/20013196/e713d517-3ba7-4a0b-a6ce-80603c9bb86c)
for example, after updating ollama to 0.1.36, then submitting a prompt to dolphin-mixtral 8x22B... in the logfile it appears to still be adding an extra zero in two places... corresponding to the two yellow-highlighted settings in OpenWebUI Settings above...
= = = logfile excerpts with my edit added to end of each line: <- comment... = = =
...
time=2024-05-11T12:17:34.323Z level=INFO source=server.go:524 msg="waiting for server to become available" status="llm server loading model"
llama_new_context_with_model: n_ctx = 60000 <- extra zero being added (bug; using OpenWebUI workaround)
...
.................................................................................................
llama_new_context_with_model: n_ctx = 20480 <- extra zero being added (bug; using OpenWebUI workaround)
